src/api.rs

```diff
@@ -86,8 +89,12 @@ pub async fn manifest(req: web::Path<ManifestRequest>, qstr: web::Query<Manifest
 	static HIT_COUNTER: Lazy<IntCounterVec> = Lazy::new(|| register_int_counter_vec!("manifest_cache_hits", "Number of manifests read from cache", &["namespace"]).unwrap());
 	static MISS_COUNTER: Lazy<IntCounterVec> = Lazy::new(|| register_int_counter_vec!("manifest_cache_misses", "Number of manifest requests that went to upstream", &["namespace"]).unwrap());
 
-	let namespace = qstr.ns.as_deref().unwrap_or_else(|| req.image.as_ref().split("/").next().unwrap_or_else(|| config.default_ns.as_ref()));
+	let mut namespace = qstr.ns.as_deref().unwrap_or_else(|| config.default_ns.as_ref());
+	if req.image.as_ref().split("/").count() > 2 {
+		namespace = req.image.as_ref().split("/").next().unwrap_or_else(|| config.default_ns.as_ref());
+	}
 
+	let image: &str = req.image.as_ref().trim_start_matches("docker.io/");
```

Comment:
Same deal with these - switch `"/"` to `'/'`

You can also avoid the mutable variable with something along these lines:
```suggestion
	let (namespace, image) = match qstr.ns.as_deref() {
		Some(ns) => (ns, req.image.as_ref()),
		None => {
			let mut parts = req.image.splitn(2, '/');
			match (parts.next(), parts.next()) {
				(Some(ns), Some(image)) => (ns, image),
				(Some(image), None) => (config.default_ns.as_ref(), image),
				(None, None) => unreachable!() // maybe want to throw an error here but this is fine for now
			}
		}
	};
```

That should work, modulo any compile errors...and giving the explicit `ns` parameter priority will prevent breaking non-`crio` deployments that have single-segment images